src/txmempool.cpp

```diff
@@ -755,11 +755,16 @@ void CTxMemPool::check(const CCoinsViewCache& active_coins_tip, int64_t spendhei
 
 bool CTxMemPool::CompareDepthAndScore(const uint256& hasha, const uint256& hashb, bool wtxid)
 {
+    /* Return `true` if hasha should be considered sooner than hashb. Namely when:
```

Comment:
For a followup it would be could to clarify "should be considered" _for what_. It's somewhat confusing to reason about because the call site reverses a and b.

The first case in the comment is "a is not in the mempool, but b is", but the line of code checks if b is not in the mempool.

Reply:
> "should be considered" for what

For processing, which begins by dropping a transaction from `m_tx_inventory_to_send` before deciding whether to send it. Since we stop processing after having sent `broadcast_max` transactions, it makes sense to do the cleaning first.

A return value of `true` means hasha is lower, i.e. would be at the bottom of the heap and processed last. But because we swap a and b when calling this, it works. That was introduced in the second commit of #7840.


We could prune them even before making a heap, but that would double the mempool lookups, and the [flame graph](https://github.com/bitcoin/bitcoin/pull/27610#issuecomment-1542339344) suggests that `make_heap` is plenty fast already now.